### PR TITLE
Conda build

### PR DESCRIPTION
**Pull request by ktbolt**
I've added the build_conda_packages.sh script that can be called from publish.sh to build moldesign conda packages.

### CODE REVIEW

File: deployment/build_conda_packages.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#
+# The latest moldesign source is obtained from the source for this repository.
+#
+# The moldesign version is obtained using the latest git tag.
+#
+# ALl conda downloads and builds are created in directories under $HOME.
+#
+cd conda
+
+# Install minconda for Python 3.6
```
Comment (avirshup) [inline deployment/build_conda_packages.sh]: Question: is this part necessary if we've already installed miniconda in the environment?

File: deployment/conda/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+pip install docker-py fortranformat 'pint>=0.7' 'pyccc>=0.6.4' 'parmed>=2.7.3'
```
Comment (avirshup) [inline deployment/conda/build.sh]: Question: what does this part do?

File: deployment/conda/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        - moldesign.orbitals
+        - moldesign.tools
+        - moldesign.units
+        - moldesign.utils
+        - moldesign.utils.docparsers
+        - moldesign.widgets
+about:
+    home: http://moldesign.bionano.autodesk.com
+    license: Apache Software License
+    license_family: Apache
```
Comment (avirshup) [inline deployment/conda/meta.yaml]: Should read "Apache 2.0"

File: deployment/build_conda_packages.sh (hunk 1 above)
Comment (ktbolt) [reply deployment/build_conda_packages.sh]: If miniconda is there then we don't need to do this. Just need to make sure that conda-build and anaconda-client are installed.

File: deployment/conda/build.sh (hunk 2 above)
Comment (ktbolt) [reply deployment/conda/build.sh]: This installs packages that are not in anaconda.

**Comment by avirshup** [comment]
@ktbolt - looks like some alignment stuff worked its way into this PR. Could you remove the changes any changes outside of the `deployment` directory from this PR?

File: deployment/build_conda_packages.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# Build package for other platforms.
+if [ $host_os == "MacOSX" ]; then
+    host_platform="osx-64"
+    other_platform="linux-64"
+else 
+    host_platform="linux-64"
+    other_platform="osx-64"
+fi
+
+conda convert $package_dir/${host_platform}/moldesign*.tar.bz2 -p ${other_platform} --output-dir $HOME/conda-packages -f
```
Comment (avirshup) [inline deployment/build_conda_packages.sh]: This is a weird one to me ... but it apparently works!

Let's go ahead and add a `linux-32` version as well, in that case ...

**Comment by avirshup** [comment]
There's some cleanup we can possibly do in the future, but I think this is good to test now. I'll create a test branch with an abbreviated CI pipeline so that I can test the conda upload step in the build environment

**Comment by ktbolt** [comment]
@avirshup I created a test-conda-upload branch with a modified codeship-steps.yml file. The build succeeded but I didn't see any files uploaded to anaconda.org.

On the code-ship build page under 'services' I see a 'publisher' tag with a rotating blue serpent biting its tail, very auspicious sign in the Hopi religion but I'm not sure what that means in code-ship land.

**Comment by avirshup** [comment]
@ktbolt - Took me a second, but the problem is:
 - The final step never ran because it only runs if the commit is tagged with a PEP440 version number
 - so, it never bothered to build the `publisher` docker image

**Comment by ktbolt** [comment]
The test-conda-upload branch CI test now works, created and upload moldesign packages to https://anaconda.org/ktbolt/moldesign.

**Pull request merged**